scripts/manila-check-affinity.py

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/env python3
+#
+# Copyright (c) 2023 SAP SE
+# All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+#
+import argparse
+import logging
+import os
+
+from helper.prometheus_exporter import LabelGauge
+from manilananny import ManilaNanny
+
+LOG = logging.getLogger(__name__)
+LOG.setLevel(logging.DEBUG)
+logging.basicConfig(level=logging.INFO, format='%(asctime)-15s %(levelname)s %(message)s')
+
+
+class ManilaCheckAffinity(ManilaNanny):
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.checks = None
+        self.affinity_groups = []
+        self.anti_affinity_groups = []
+        self.affinityGauge = LabelGauge("manila_nanny_affinity_rule_violation",
+                                        "Violated Affinity Rules", ['share_id', 'rule'])
+        self.antiAffinityGauge = LabelGauge("manila_nanny_anti_affinity_rule_violation",
+                                            "Violated Anti Affinity Rules", ['share_id', 'rule'])
+
+    def _run(self):
+        LOG.info("Checking Anti-Affinity Rules")
+
+        self._update_anti_affinity_groups()
+
+        violated_rules = []
+        for grp in self.anti_affinity_groups:
+            check, share_hosts = self._check_group(grp, attracting=False)
+            if not check:
+                LOG.error("Anti-Affinity rule violated for Shares")
+                for s in share_hosts:
+                    LOG.error("Share %s on host %s", s['share_id'], s['host'])
```

Comment:
violated_rules need to be filled in here, otherwise the antiAffinityGauge is always empty